compiler/qsc_codegen/src/qir.rs

```diff
@@ -546,9 +544,92 @@ impl ToQir<String> for rir::Program {
         } else {
             "adaptive_profile"
         };
-        format!(
+        let body = format!(
             include_str!("./qir/template.ll"),
             callables, profile, self.num_qubits, self.num_results
-        )
+        );
+        let flags = get_module_metadata(self);
+        body + "\n" + &flags
+    }
+}
+
+/// Create the module metadata for the given program.
+/// creating the `llvm.module.flags` and its associated values.
+fn get_module_metadata(program: &rir::Program) -> String {
+    let mut flags = String::new();
+
+    // push the default attrs, we don't have any config values
+    // for now that would change any of them.
+    flags.push_str(
+        r#"
+!0 = !{i32 1, !"qir_major_version", i32 1}
+!1 = !{i32 7, !"qir_minor_version", i32 0}
+!2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+!3 = !{i32 1, !"dynamic_result_management", i1 false}
+"#,
+    );
+
+    let mut index = 4;
+
+    // If we are not in the base profile, we need to add the capabilities
+    // associated with the adaptive profile.
+    if !program.config.is_base() {
+        // loop through the capabilities and add them to the metadata
+        // for values that we can generate.
+        for cap in program.config.capabilities.iter() {
+            let name = match cap {
```

Comment:
nitpick: since this match expression is the same as the one below, I wonder if there is reason to have a utlity to turn capability flags into the matching QIR attributes. Not critical by any means.